mobile/src/main/java/org/openhab/habdroid/util/ExtensionFuncs.kt

```diff
@@ -177,6 +177,11 @@ inline fun <T> JSONArray.map(transform: (JSONObject) -> T): List<T> {
     return (0 until length()).map { index -> transform(getJSONObject(index)) }
 }
 
+fun JSONObject.getStringOrNull(key: String): String? {
```

Comment:
While this fixes all the warnings, I really don't like the solution. Why must be the fallback value in optString non-null? :|

Reply:
We can't change that anyway, so thinking about that rationale is pointless ;-)
I actually like the solution, but the method needs to be called `optStringOrNull`, since the get* methods are for fetching *mandatory* fields (JsonException is thrown if they aren't present).